backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ir/NewIrUtils.kt

```diff
@@ -17,24 +17,29 @@ import org.jetbrains.kotlin.ir.expressions.IrConstructorCall
 import org.jetbrains.kotlin.ir.expressions.impl.IrConstImpl
 import org.jetbrains.kotlin.ir.expressions.impl.IrConstructorCallImpl
 import org.jetbrains.kotlin.ir.symbols.IrClassSymbol
+import org.jetbrains.kotlin.ir.types.IdSignatureValues
 import org.jetbrains.kotlin.ir.types.classifierOrFail
 import org.jetbrains.kotlin.ir.types.isMarkedNullable
 import org.jetbrains.kotlin.ir.util.*
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.name.Name
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-val IrField.fqNameForIrSerialization: FqName get() = this.parent.fqNameForIrSerialization.child(this.name)
+private fun IrClass.isClassTypeWithSignature(signature: IdSignature.PublicSignature): Boolean {
+    if (!symbol.isPublicApi || !symbol.signature.isPublic) return false
```

Comment:
Why is this required?

Reply:
Symbols which isn't public don't have signature, there is an error inside `signature` implementation for private classes.

Reply:
I see, makes sense.
But this explains only `!symbol.isPublicApi` part. What is `!symbol.signature.isPublic` required for?

Reply:
Also, the entire pattern `if (symbol.isPublic) symbol.signature` looks like an unsafe leaking abstraction to me.
More safe and encapsulated solution would probably be to define an extension property like `symbol.publicSignatureOrNull` or `symbol.signatureOrNull` near the `IrSymbol` definition.
Need an opinion from @udalov on this though.

Reply:
> !symbol.signature.isPublic

It was made for checking that info is consistent inside class, I can remove it without any problems I think.

> symbol.publicSignatureOrNull or symbol.signatureOrNull near the IrSymbol definition.

Good point! Thank you.

Reply:
My opinion is that we should remove `isPublic` and make `signature` nullable, it would simplify everything. But not in these changes of course.

Reply:
> make signature nullable

If so I wouldn't make now it as separate function and just fix this when signature will be nullable.